java-connectors/kafka-connect-gcp-common/src/test/java/io/lenses/streamreactor/connect/gcp/common/auth/GCPServiceBuilderConfigurerTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.gcp.common.auth;
+
+
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.cloud.*;
+import com.google.cloud.http.HttpTransportOptions;
+import io.lenses.streamreactor.common.config.base.RetryConfig;
+import io.lenses.streamreactor.connect.gcp.common.auth.mode.NoAuthMode;
+import lombok.val;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.*;
+
+class GCPServiceBuilderConfigurerTest {
+
+    private GCPConnectionConfig.GCPConnectionConfigBuilder configBuilder;
+
+    @BeforeEach
+    public void setUp() {
+        configBuilder = GCPConnectionConfig.builder()
+                .projectId(Optional.empty())
```

Comment:
this unfortunately defeats the purpose of using `@Builder` 😔

Reply:
💯